.github/workflows/macos.yml

```diff
@@ -42,28 +41,40 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      - name:  Prepare brew and compiler caches
+        id:    prep-caches
+        shell: bash
+        run: |
+          set -eu
+          BREW_DIR="$(brew --cache)"
+          DISCARD_DIR="${{ github.workspace }}/discard"
+          mkdir -p "$DISCARD_DIR"
+          mv -f "$BREW_DIR"/* "$DISCARD_DIR"
+          mkdir -p "$CCACHE_DIR"
+          echo "::set-output name=brew_dir::$BREW_DIR"
+          echo "::set-output name=ccache_dir::$CCACHE_DIR"
+          echo "::set-output name=today::$(date +%F)"
+          echo "::set-output name=name_hash::$(echo '${{ matrix.conf.name }}' | shasum | cut -b-8)"
+
+      - uses:  actions/cache@v2
+        with:
+          path: ${{ steps.prep-caches.outputs.brew_dir }}
+          key:  brew-cache-${{ steps.prep-caches.outputs.today }}
+          restore-keys: brew-cache-
+
       - name: Install C++ compiler and libraries
         run: |
           brew install \
             ${{ matrix.conf.packages }} \
             $(cat ./.github/packages/macos-latest-brew.txt)
 
-      - name:  Prepare compiler cache
-        id:    prep-ccache
-        shell: bash
-        run: |
-          mkdir -p "${CCACHE_DIR}"
-          echo "::set-output name=dir::$CCACHE_DIR"
-          echo "::set-output name=today::$(gdate -I)"
-          echo "::set-output name=yesterday::$(gdate --date=yesterday -I)"
-
       - uses:  actions/cache@v2
-        id:    cache-ccache
         with:
-          path: ${{ steps.prep-ccache.outputs.dir }}
-          key:  ccache-${{ matrix.conf.os }}-${{ steps.prep-ccache.outputs.today }}
+          path: ${{ steps.prep-caches.outputs.ccache_dir }}
+          key:  ccache-macos-debug-${{ steps.prep-caches.outputs.name_hash }}-${{ steps.prep-caches.outputs.today }}
```

Comment:
The `name_hash` adds uniqueness to the key so that each build-type gets its own ccache'd archive. 

(background on why I used the 8-value hash of `matrix.conf.name`instead of `name` directly is that the GitHub key doesn't accept some of the charactes used in our name strings - like commas).

Reply:
> (background on why I used the 8-value hash of matrix.conf.nameinstead of name directly is that the GitHub key doesn't accept some of the charactes used in our name strings - like commas).

We shouldn't use space-separated names for cache names anyway - good change :)